adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java

```diff
@@ -71,6 +79,25 @@ public void deleteAllDatasources() {
     StreamSupport.stream(allDatasourceConfigs.spliterator(), true).forEach(amqpPublisher::publishDeletion);
   }
 
+  public void validate(Datasource datasource, DataImport dataImport) throws IOException, ValidationException {   
+    try 
+    {
+      String schemaString = new Gson().toJson(datasource.getSchema());
+      JSONObject rawSchema = new JSONObject(schemaString);
+      Schema schema = SchemaLoader.load(rawSchema);
+      String dataString = dataImport.getData();
+      if (dataString.substring(0, 1).equals("[")) {
+        schema.validate(new JSONArray(dataImport.getData()));
+      }
+      else {
+        schema.validate(new JSONObject(dataImport.getData()));
+      }
+    } catch ( ValidationException e) {
+      throw e;
+    }
+    
+  }
+
```

Comment:
Please add some unit tests for the validation part in the dataImport (1 test for success, 1 for warning, 1 for without schema, 1 for error)